Source/Plugin.BLE/Android/BroadcastReceivers/BondStatusBroadcastReceiver.cs

```diff
@@ -1,41 +1,55 @@
 using System;
 using Android.Bluetooth;
 using Android.Content;
-using Plugin.BLE.Abstractions;
+using Android.OS;
 using Plugin.BLE.Abstractions.EventArgs;
 using Plugin.BLE.Android;
+using Plugin.BLE.Extensions;
 
 namespace Plugin.BLE.BroadcastReceivers
 {
-    //[BroadcastReceiver]
     public class BondStatusBroadcastReceiver : BroadcastReceiver
     {
+	    private readonly Adapter _broadcastAdapter;
+
         public event EventHandler<DeviceBondStateChangedEventArgs> BondStateChanged;
 
+        public BondStatusBroadcastReceiver(Adapter adapter)
+        {
+            _broadcastAdapter = adapter;
+        }
+
         public override void OnReceive(Context context, Intent intent)
         {
-            var bondState = (Bond)intent.GetIntExtra(BluetoothDevice.ExtraBondState, (int)Bond.None);
-            //ToDo
-            var device = new Device(null, (BluetoothDevice)intent.GetParcelableExtra(BluetoothDevice.ExtraDevice), null);
-            Console.WriteLine(bondState.ToString());
+            if (BondStateChanged == null)
+            {
+	            return;
+            }
 
-            if (BondStateChanged == null) return;
+            var extraBondState = (Bond)intent.GetIntExtra(BluetoothDevice.ExtraBondState, (int)Bond.None);
 
-            switch (bondState)
+            BluetoothDevice bluetoothDevice;
+            
+            // As older versions of .NET (pre7) don't include the Tiramisu version code, handle it manually.
+#if NET6_0_OR_GREATER
+	        if (OperatingSystem.IsAndroidVersionAtLeast(33))
+#else
+            if ((int)Build.VERSION.SdkInt >= 33)
+#endif
```

Comment:
With #740 merged, you should also be able to use the Tiramisu version code here (instead of casting to int). Then the comment in line 33 can go away.
Also: The whitespace/indentation in line 35 looks odd.

Reply:
Ah, somehow a Tab sneaked in. I blame Rider 😄 

Changed the cast to BuildVersionCode.Tiramisu.   
Also fixed the indentation 🙈

Reply:
Awesome. I think this block of code is finally where it should be. Thanks 😁